File: flutter/packages/measure_flutter/test/tracing/msr_span_builder_test.dart
import 'package:flutter_test/flutter_test.dart';
import 'package:measure_flutter/src/time/time_provider.dart';
import 'package:measure_flutter/src/tracing/msr_span_builder.dart';

import '../utils/fake_id_provider.dart';
import '../utils/fake_span_processor.dart';
import '../utils/fake_trace_sampler.dart';
import '../utils/noop_logger.dart';
import '../utils/test_clock.dart';

void main() {
  final logger = NoopLogger();
  final testClock = TestClock.create();
  final timeProvider = FlutterTimeProvider(testClock);
  final spanProcessor = FakeSpanProcessor();
  final sampler = FakeSampler();
  final idProvider = FakeIdProvider();

  group("MsrSpanBuilder", () {
    test('setsParent sets span parent', () {
      final parentSpan = MsrSpanBuilder(
        name: 'parent-name',
        idProvider: idProvider,
        timeProvider: timeProvider,
        spanProcessor: spanProcessor,
        sampler: sampler,
        logger: logger,
      ).startSpan();

      final span = MsrSpanBuilder(
        name: 'span-name',
        idProvider: idProvider,
        timeProvider: timeProvider,
        spanProcessor: spanProcessor,
        sampler: sampler,
        logger: logger,
      ).setParent(parentSpan).startSpan();

      expect(span.parentId, parentSpan.spanId);
    });
  });
}

File: android/measure-android/measure/src/test/java/sh/measure/android/tracing/MsrSpanBuilderTest.kt
package sh.measure.android.tracing

import org.junit.Assert
import org.junit.Test
import sh.measure.android.fakes.FakeSampler
import sh.measure.android.fakes.FakeSessionManager
import sh.measure.android.fakes.NoopLogger
import sh.measure.android.fakes.NoopSpanProcessor
import sh.measure.android.utils.AndroidTimeProvider
import sh.measure.android.utils.IdProviderImpl
import sh.measure.android.utils.RandomizerImpl
import sh.measure.android.utils.TestClock

class MsrSpanBuilderTest {
    private val idProvider = IdProviderImpl(randomizer = RandomizerImpl())
    private val testClock = TestClock.create()
    private val timeProvider = AndroidTimeProvider(testClock)
    private val logger = NoopLogger()
    private val spanProcessor = NoopSpanProcessor()
    private val sessionManager = FakeSessionManager()
    private val sampler = FakeSampler()

    @Test
    fun `setsParent sets span parent`() {
        val parentSpan = MsrSpanBuilder(
            "parent-name",
            idProvider,
            timeProvider,
            spanProcessor,
            sessionManager,
            sampler,
            logger,
        ).startSpan()
        val span = MsrSpanBuilder(
            "span-name",
            idProvider,
            timeProvider,
            spanProcessor,
            sessionManager,
            sampler,
            logger,
        ).setParent(parentSpan).startSpan()

        Assert.assertEquals(parentSpan.spanId, span.parentId)
    }
}
